packages/rich-text-markdown-renderer/src/index.ts

```diff
@@ -0,0 +1,88 @@
+import he from 'he';
+import { BLOCKS, MARKS, INLINES, Block, Inline, Document } from '@contentful/rich-text-types';
+import { documentToHtmlString } from '@contentful/rich-text-html-renderer';
+
+export interface RenderNode {
+  [k: string]: NodeRenderer;
+}
+
+export interface RenderMark {
+  [k: string]: (text: string) => string;
+}
+
+export type CommonNode = Text | Block | Inline;
+
+export interface Next {
+  (nodes: CommonNode[]): string;
+}
+
+export interface NodeRenderer {
+  (node: Block | Inline, next: Next): string;
+}
+
+export interface Options {
+  /**
+   * Node renderers
+   */
+  renderNode?: RenderNode;
+  /**
+   * Mark renderers
+   */
+  renderMark?: RenderMark;
+}
+
+export function documentToMarkdown(document: Document, options: Partial<Options> = {}): string {
+  let orderedListCounter = -1;
+  let nestedListCounter = -1;
+
+  return he.decode(
+    documentToHtmlString(document, {
+      renderMark: {
+        [MARKS.BOLD]: (text: any) => `**${text}**`,
+        [MARKS.ITALIC]: (text: any) => `*${text}*`,
+        [MARKS.UNDERLINE]: (text: any) => `~~${text}~~`,
+        [MARKS.CODE]: (text: any) => `\`${text}\``,
```

Comment:
Does markdown support superscript and subscript?

Reply:
Yes, it indeed supports `sub` and `sup` but by using the HTML tags.

I'm not sure if we should try to address this here or leave it to users since it's up to the users if they want to support HTML tags